lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -775,4 +803,95 @@ defmodule Trento.Domain.SapSystem do
       }
     end
   end
+
+  defp maybe_emit_sap_system_deregistered_event(
+         %SapSystem{sid: nil},
+         _deregistered_at
+       ),
+       do: []
+
+  defp maybe_emit_sap_system_deregistered_event(
```

Comment:
I think we are missing one scenario, the `database: %{sid: nil}`.
In fact, we shouldn't check for instances to be an empty list, we should check for the above condition.
We can have one secondary instance but a deregistered database.
Right?